### PR TITLE
[Merged by Bors] - feat(Algebra/Module/Injective): injective modules are Baer

### PR DESCRIPTION
**Pull request by jjaassoonn**
We already know that Baer modules are injective, the other direction is easier

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/Module/Injective.lean [hunk 1]
```diff
@@ -465,4 +465,10 @@ protected theorem injective (h : Module.Baer R Q) : Module.Injective R Q :=
 set_option linter.uppercaseLean3 false in
 #align module.Baer.injective Module.Baer.injective
 
+protected theorem ofInjective (inj : Module.Injective R Q) : Module.Baer R Q := fun I g ↦
+  let ⟨g', hg'⟩ := inj.1 (ULift.{max u v} I) (ULift.{max u v} R)
+    (⟨⟨ULift.up, by aesop⟩, by aesop⟩ ∘ₗ I.subtype ∘ₗ ⟨⟨ULift.down, by aesop⟩, by aesop⟩)
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Injective.lean:470]: I think `⟨⟨ULift.up, by aesop⟩, by aesop⟩` is better spelt `ULift.moduleEquiv` or similar.

Comment (jjaassoonn) [reply Mathlib/Algebra/Module/Injective.lean]: done

File: Mathlib/Algebra/Module/Injective.lean [hunk 2]
```diff
@@ -465,4 +466,10 @@ protected theorem injective (h : Module.Baer R Q) : Module.Injective R Q :=
 set_option linter.uppercaseLean3 false in
 #align module.Baer.injective Module.Baer.injective
 
+protected theorem ofInjective (inj : Module.Injective R Q) : Module.Baer R Q := fun I g ↦
+  let ⟨g', hg'⟩ := inj.1 (ULift.{max u v} I) (ULift.{max u v} R)
+    (ULift.moduleEquiv.symm.toLinearMap ∘ₗ I.subtype ∘ₗ ULift.moduleEquiv.toLinearMap)
+    (fun a b h ↦ by aesop) (g ∘ₗ ULift.moduleEquiv.toLinearMap)
+  ⟨g' ∘ₗ ULift.moduleEquiv.symm.toLinearMap, by aesop⟩
+
```
Comment (eric-wieser) [inline Mathlib/Algebra/Module/Injective.lean:474]: Can you add the `Iff`?

**Comment by jjaassoonn** [comment]
I lost my computer, I am just eyeballing here

File: Mathlib/Algebra/Module/Injective.lean (hunk 2 above)
Comment (jjaassoonn) [reply Mathlib/Algebra/Module/Injective.lean:474]: Done

**Review by alreadydone — APPROVED** [review]
maintainer merge

**Comment by RemyDegenne** [comment]
Thanks!
bors r+

**Pull request closed without merging**